### PR TITLE
Added the Padding and Activation scenarios

### PR DESCRIPTION
**Pull request by amitsrivastava78**
This was one of the TODO items

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @aselle: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/lite/kernels/pooling_test.cc [hunk 1]
```diff
@@ -26,8 +26,21 @@ using ::testing::ElementsAreArray;
 
 class BasePoolingOpModel : public SingleOpModel {
  public:
-  // TODO(ahentz): Also test different activation types, bias, padding types,
-  // stride values.
+  BasePoolingOpModel(BuiltinOperator type, const TensorData& input,
```
Comment (haozha111) [inline tensorflow/lite/kernels/pooling_test.cc]: For function parameters, could you use const reference instead of passing-by-value here?

Comment (amitsrivastava78) [reply tensorflow/lite/kernels/pooling_test.cc]: @haozha111 , thanks for pointing this out, i have made the changes as suggested by you, kindly check and approve.

**Comment by amitsrivastava78** [comment]
@haozha111 , thanks for pointing this out, i have made the changes as suggested by you, kindly check and approve.

File: tensorflow/lite/kernels/pooling_test.cc [hunk 2]
```diff
@@ -26,8 +26,22 @@ using ::testing::ElementsAreArray;
 
 class BasePoolingOpModel : public SingleOpModel {
  public:
-  // TODO(ahentz): Also test different activation types, bias, padding types,
-  // stride values.
+  BasePoolingOpModel(const BuiltinOperator& type, const TensorData& input,
+                     const int& filter_width, const int& filter_height,
```
Comment (haozha111) [inline tensorflow/lite/kernels/pooling_test.cc]: It's fine to use: int filter_width, int filter_height instead.

Comment (amitsrivastava78) [reply tensorflow/lite/kernels/pooling_test.cc]: @haozha111 Sure, made the change as per the comments, kindly check and approve

**Comment by amitsrivastava78** [comment]
@haozha111 Sure, made the change as per the comments, kindly check and approve

File: tensorflow/lite/kernels/pooling_test.cc [hunk 3]
```diff
@@ -26,8 +26,22 @@ using ::testing::ElementsAreArray;
...
+  BasePoolingOpModel(const BuiltinOperator& type, const TensorData& input,
+                     int filter_width, int filter_height,
+                     const Padding& padding,
+                     const ActivationFunctionType& fused_activation_function,
+                     const TensorData& output) {
+    input_ = AddInput(input);
+    output_ = AddOutput(output);
+
+    SetBuiltinOp(type, BuiltinOptions_Pool2DOptions,
+                 CreatePool2DOptions(builder_, padding, 2, 2, filter_width,
```
Comment (haozha111) [inline tensorflow/lite/kernels/pooling_test.cc]: It seems that strides are hard-coded here. Could you make it configurable and make a unit test on strides also?

Comment (amitsrivastava78) [reply tensorflow/lite/kernels/pooling_test.cc]: @haozha111 , thanks for the feedback, i have updated the code as per your suggestion. Kindly check and approve.

**Comment by amitsrivastava78** [comment]
@haozha111 , thanks for the feedback, i have updated the code as per your suggestion. Kindly check and approave.

File: tensorflow/lite/kernels/pooling_test.cc [hunk 4]
```diff
@@ -26,8 +26,22 @@ using ::testing::ElementsAreArray;
...
+  BasePoolingOpModel(const BuiltinOperator& type, const TensorData& input,
+                     int filter_width, int filter_height,
+                     const Padding& padding,
+                     const ActivationFunctionType& fused_activation_function,
+                     const TensorData& output) {
+    input_ = AddInput(input);
+    output_ = AddOutput(output);
+
+    SetBuiltinOp(type, BuiltinOptions_Pool2DOptions,
+                 CreatePool2DOptions(builder_, padding, filter_width, filter_height, filter_width,
```
Comment (haozha111) [inline tensorflow/lite/kernels/pooling_test.cc]: a little typo here: the first filter_width and filter_hight should be stride_width, and stride_height.

Comment (amitsrivastava78) [reply tensorflow/lite/kernels/pooling_test.cc]: @haozha111 , sorry for the overlook, i have fixed the issue, kindly check and approve.

**Comment by amitsrivastava78** [comment]
@haozha111 , sorry for the overlook, i have fixed the issue, kindly check and approve.

File: tensorflow/lite/kernels/pooling_test.cc [hunk 5]
```diff
@@ -12,8 +12,8 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-#include <cstdarg>
 #include <gtest/gtest.h>
+#include <cstdarg>
```
Comment (haozha111) [inline tensorflow/lite/kernels/pooling_test.cc]: please restore the header import order.

File: tensorflow/lite/kernels/pooling_test.cc [hunk 6]
```diff
@@ -26,8 +26,23 @@ using ::testing::ElementsAreArray;
...
+  BasePoolingOpModel(const BuiltinOperator& type, const TensorData& input,
+                     int stride_width, int stride_height, int filter_width,
+                     int filter_height, const Padding& padding,
+                     const ActivationFunctionType& fused_activation_function,
+                     const TensorData& output) {
+    input_ = AddInput(input);
+    output_ = AddOutput(output);
+
+    SetBuiltinOp(type, BuiltinOptions_Pool2DOptions,
+                 CreatePool2DOptions(builder_, padding, filter_width,
```
Comment (haozha111) [inline tensorflow/lite/kernels/pooling_test.cc]: CreatePool2DOptions(builder_, padding, stride_width,
                                     stride_height, filter_width, filter_height,
                                     fused_activation_function)

File: tensorflow/lite/kernels/pooling_test.cc (hunk 5 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/pooling_test.cc]: @haozha111 , Sure i have restored the order.

File: tensorflow/lite/kernels/pooling_test.cc (hunk 6 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/pooling_test.cc]: @haozha111 , added stride_width,stride_height as passed from the function.

**Comment by amitsrivastava78** [comment]
@haozha111 , I have updated the code as per the suggestions, kindly check and approve,

Regards
Amit

**Review by haozha111 — APPROVED** [review]

**Pull request merged**